flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NetworkEnvironmentConfiguration.java

```diff
@@ -105,6 +106,300 @@ public NettyConfig nettyConfig() {
 		return nettyConfig;
 	}
 
+	public boolean isCreditBased() {
+		return isCreditBased;
+	}
+
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Utility method to extract network related parameters from the configuration and to
+	 * sanity check them.
+	 *
+	 * @param configuration configuration object
+	 * @param maxJvmHeapMemory the maximum JVM heap size (in bytes)
+	 * @param localTaskManagerCommunication true, to skip initializing the network stack
+	 * @param taskManagerAddress identifying the IP address under which the TaskManager will be accessible
+	 * @return NetworkEnvironmentConfiguration
+	 */
+	@Deprecated
+	public static NetworkEnvironmentConfiguration fromConfiguration(
```

Comment:
Because the `@Deprecated` `TaskManagerOptions.NETWORK_NUM_BUFFERS` is used in this method.

Reply:
True, then we should probably use `@SuppressWarnings("deprecation")` in this case because the method `fromConfiguration` itself does not look deprecated.